### ISSUE TITLE
Consume quick fixes for LSP4MP

### ISSUE THREAD

**Issue by TrevCraw** [OpenLiberty/liberty-tools-intellij#122]
Find all the quick fix Java files and make list in this issue. Rank in order of difficulty and sort by similarity. Start with the easier ones first.
Consider quick fixes for both Java and properties files.
dependency on: https://github.com/OpenLiberty/liberty-tools-intellij/issues/81
Need: org.eclipse.lsp4mp.jdt.internal.core.java.codeaction.CodeActionHandler

List of Eclipse LSP4MP quick-fix classes (to be Ranked):

Fixed in #320
- [x] ApplicationScopedAnnotationMissingQuickFix
- [x] HealthAnnotationMissingQuickFix
- [x] InsertAnnotationMissingQuickFix
- [x] MicroProfileGenerateOpenAPIOperation
- [x] NoValueAssignedToPropertyQuickFix - need to make sure project has a `META-INF/microprofile-config.properties` file. the MP starter project did not have this file in the META-INF directory

Fixed in #286 
- [x] ImplementHealthCheckQuickFix
- [x] InsertAnnotationAttributeQuickFix

**Comment by dshimo** [comment]
The `plugin.xml` helps pair a diagnostic to its associated quick fix.

```
    <javaCodeActionParticipant kind="quickfix"
            targetDiagnostic="microprofile-config#NO_VALUE_ASSIGNED_TO_PROPERTY"
            implementationClass="io.openliberty.tools.intellij.lsp4mp4ij.psi.internal.config.java.NoValueAssignedToPropertyQuickFix"/>
    <javaCodeActionParticipant kind="quickfix"
            targetDiagnostic="microprofile-config#NO_VALUE_ASSIGNED_TO_PROPERTY"
            implementationClass="io.openliberty.tools.intellij.lsp4mp4ij.psi.internal.config.java.InsertDefaultValueAnnotationAttributeQuickFix"/>
    <javaCodeActionParticipant kind="quickfix"
            targetDiagnostic="microprofile-health#ImplementHealthCheck"
            implementationClass="io.openliberty.tools.intellij.lsp4mp4ij.psi.internal.health.java.ImplementHealthCheckQuickFix"/>
    <javaCodeActionParticipant kind="quickfix"
            targetDiagnostic="microprofile-health#HealthAnnotationMissing"
            implementationClass="io.openliberty.tools.intellij.lsp4mp4ij.psi.internal.health.java.HealthAnnotationMissingQuickFix"/>
    <javaCodeActionParticipant kind="quickfix"
            targetDiagnostic="microprofile-metrics#ApplicationScopedAnnotationMissing"
            implementationClass="io.openliberty.tools.intellij.lsp4mp4ij.psi.internal.metrics.java.ApplicationScopedAnnotationMissingQuickFix"/>
    <javaCodeActionParticipant kind="quickfix"
            targetDiagnostic="microprofile-restclient#InjectAnnotationMissing"
            implementationClass="io.openliberty.tools.intellij.lsp4mp4ij.psi.internal.restclient.java.InjectAnnotationMissingQuickFix"/>
    <javaCodeActionParticipant kind="quickfix"
            targetDiagnostic="microprofile-restclient#RestClientAnnotationMissing"
            implementationClass="io.openliberty.tools.intellij.lsp4mp4ij.psi.internal.restclient.java.RestClientAnnotationMissingQuickFix"/>
    <javaCodeActionParticipant kind="quickfix"
            targetDiagnostic="microprofile-restclient#InjectAndRestClientAnnotationMissing"
            implementationClass="io.openliberty.tools.intellij.lsp4mp4ij.psi.internal.restclient.java.InjectAndRestClientAnnotationMissingQuickFix"/>
    <javaCodeActionParticipant kind="quickfix"
            targetDiagnostic="microprofile-restclient#RegisterRestClientAnnotationMissing"
            implementationClass="io.openliberty.tools.intellij.lsp4mp4ij.psi.internal.restclient.java.RegisterRestClientAnnotationMissingQuickFix"/>
```

It seems so far that the common fail point is an out of bounds exception: 
`java.lang.IndexOutOfBoundsException: Wrong line: 27. Available lines count: 26`
The running theory assumes that document changes aren't being registered properly and that a quick fix 'timing' is off. As a vaguely similar symptom of doc de-sync, we can hopefully resolve these issues together with a proper #287 fix

**Comment by evie-lau** [comment]
Regarding the `IndexOutOfBoundsException`, seems to be due to using the Range from the proposed edited new text, but on the original document which may not contain the same number of lines.
This particular exception be fixed by #320

**Comment by evie-lau** [comment]
Cannot seem to find a diagnostic which provides a quickfix using `NoValueAssignedToPropertyQuickFix`
